task/sql-tasks.js

```diff
@@ -280,7 +448,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    select distinct 
+        c.CompanyName as "CompanyName",
+	    p.ProductName as "ProductName",
+	    od.UnitPrice as "PricePerItem"
+    from Products p 
+    join OrderDetails od on p.ProductID = od.ProductID 
+    join Orders o on od.OrderID = o.OrderID 
+    join Customers c on o.CustomerID = c.CustomerID
+    where od.UnitPrice = (
+        select 
+            max(od2.UnitPrice )
+        from Customers c1
+        join Orders o2 on o2.CustomerID = c1.CustomerID 
+        join OrderDetails od2 on o2.OrderID = od2.OrderID 
+        where c1.CompanyName = c.CompanyName 

```

Comment:
according to the schema company name doesn't uniquely identify customer. It's also slow because there no index on that field